### PR TITLE
EOS-27069: hare does not support multiple network transports

### PR DESCRIPTION
**Pull request by mssawant**
Presently Hare supports just libfabric in Containers. This restricts testing
Hare and Motr using different transports, e.g. lnet.

Solution:
- Update Hare configuration to support multiple transports.

Signed-off-by: Mandar Sawant <mandar.sawant@seagate.com>

### CODE REVIEW

File: cfgen/cfgen [hunk 1]
```diff
@@ -1073,16 +1077,19 @@ class ConfProcess(ToDhall):
...
-        # ep = Endpoint(proto=NetProtocol[proto],
-        #               ipaddr=facts[ipaddr_key(node_desc['data_iface'])],
-        #              portal=proc_t.value)
-        ep = LibfabricEndpoint(NetFamily='inet', proto=NetProtocol[proto],
-                               ipaddr=facts[ipaddr_key(
-                                            node_desc['data_iface'])],
-                               portalgroup=proc_t.value)
+        transport_type = node_desc.get('transport_type', 'libfab')
+        ep: Any = (
+            LibfabricEndpoint(NetFamily='inet', proto=NetProtocol[proto],
```
Comment (madhavemuri) [inline cfgen/cfgen:1082]: Consider this as a generic endpoint,
Transport_type can also be of type "sock" which too follows the same format as that of libfabric.

**Comment by SwapnilGaonkar7** [comment]
@mssawant Why do we need separate 'update-consul-conf' file for 'libfab'?

File: cfgen/cfgen (hunk 1 above)
Comment (mssawant) [reply cfgen/cfgen:1082]: Yeah a new endpoint format can be added easily, just need to add it to the union and add a corresponding rendering function. Can do this once we enable sock transport.

**Comment by mssawant** [comment]
>Why do we need separate 'update-consul-conf' file for 'libfab'?

@SwapnilGaonkar7, lnet endpoint parsing is different than libfabric endpoint parsing. Having multilpe files is better than complicating the existing update-consul-conf script. We should later change update-consul-conf functionality into a python script. In in present situation, I think having separate files is cleaner to maintain.

File: cfgen/cfgen (hunk 1 above)
Comment (mssawant) [reply cfgen/cfgen:1082]: Having specific static typing helps to implement strict typing during configuration and avoid any unwanted and dynamic changes to the formats.

**Comment by mssawant** [comment]
@SwapnilGaonkar7, on second thought, since update-consul-conf is referred in other multiple scripts and there are not much difference in the parsing code with respect to parsing libfab and lnet, I update the patch by reverting to a single update-consul-conf and added `xprt` as a parameter to the related scripts.

**Review by vaibhavparatwar — APPROVED** [review]

**Review by SwapnilGaonkar7 — APPROVED** [review]

**Comment by mssawant** [comment]
Deployment was successful for the PR at http://eos-jenkins.colo.seagate.com/job/Cortx-kubernetes/job/deploy-cortx-k8-cluster/651/console
```
[root@cortx-data-headless-svc-ssc-vm-g2-rhev4-1473 /]# hctl status
Data pool:
    # fid name
    0x6f00000000000001:0x24f 'storage-set-1__sns'
Profile:
    # fid name: pool(s)
    0x7000000000000001:0x381 'Profile_the_pool': 'storage-set-1__sns' 'storage-set-1__dix' None
Services:
    cortx-data-headless-svc-ssc-vm-g3-rhev4-0816  (RC)
    [started]  hax        0x7200000000000001:0x12d  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@2001
    [started]  ioservice  0x7200000000000001:0x130  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@3001
    [started]  ioservice  0x7200000000000001:0x13d  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@3002
    [started]  confd      0x7200000000000001:0x14a  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@3003
    [started]  s3server   0x7200000000000001:0x14d  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@4001
    [started]  s3server   0x7200000000000001:0x150  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0816@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-2237
    [started]  hax        0x7200000000000001:0x157  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@2001
    [started]  ioservice  0x7200000000000001:0x15a  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@3001
    [started]  ioservice  0x7200000000000001:0x167  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@3002
    [started]  confd      0x7200000000000001:0x174  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@3003
    [started]  s3server   0x7200000000000001:0x177  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@4001
    [started]  s3server   0x7200000000000001:0x17a  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2237@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1631
    [started]  hax        0x7200000000000001:0x181  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@2001
    [started]  ioservice  0x7200000000000001:0x184  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@3001
    [started]  ioservice  0x7200000000000001:0x191  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@3002
    [started]  confd      0x7200000000000001:0x19e  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@3003
    [started]  s3server   0x7200000000000001:0x1a1  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@4001
    [started]  s3server   0x7200000000000001:0x1a4  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1631@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1632
    [started]  hax        0x7200000000000001:0x1ab  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@2001
    [started]  ioservice  0x7200000000000001:0x1ae  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@3001
    [started]  ioservice  0x7200000000000001:0x1bb  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@3002
    [started]  confd      0x7200000000000001:0x1c8  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@3003
    [started]  s3server   0x7200000000000001:0x1cb  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@4001
    [started]  s3server   0x7200000000000001:0x1ce  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1632@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1635
    [started]  hax        0x7200000000000001:0x1d5  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@2001
    [started]  ioservice  0x7200000000000001:0x1d8  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@3001
    [started]  ioservice  0x7200000000000001:0x1e5  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@3002
    [started]  confd      0x7200000000000001:0x1f2  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@3003
    [started]  s3server   0x7200000000000001:0x1f5  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@4001
    [started]  s3server   0x7200000000000001:0x1f8  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1635@4002
    cortx-data-headless-svc-ssc-vm-g3-rhev4-0877
    [started]  hax        0x7200000000000001:0x1ff  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@2001
    [started]  ioservice  0x7200000000000001:0x202  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@3001
    [started]  ioservice  0x7200000000000001:0x20f  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@3002
    [started]  confd      0x7200000000000001:0x21c  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@3003
    [started]  s3server   0x7200000000000001:0x21f  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@4001
    [started]  s3server   0x7200000000000001:0x222  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-0877@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1473
    [started]  hax        0x7200000000000001:0x229  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@2001
    [started]  ioservice  0x7200000000000001:0x22c  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@3001
    [started]  ioservice  0x7200000000000001:0x239  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@3002
    [started]  confd      0x7200000000000001:0x246  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@3003
    [started]  s3server   0x7200000000000001:0x249  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@4001
    [started]  s3server   0x7200000000000001:0x24c  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1473@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-2238
    [started]  hax        0x7200000000000001:0x31  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@2001
    [started]  ioservice  0x7200000000000001:0x34  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@3001
    [started]  ioservice  0x7200000000000001:0x41  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@3002
    [started]  confd      0x7200000000000001:0x4e  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@3003
    [started]  s3server   0x7200000000000001:0x51  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@4001
    [started]  s3server   0x7200000000000001:0x54  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-2238@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1630
    [started]  hax        0x7200000000000001:0x7   inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@2001
    [started]  ioservice  0x7200000000000001:0xa   inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@3001
    [started]  ioservice  0x7200000000000001:0x17  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@3002
    [started]  confd      0x7200000000000001:0x24  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@3003
    [started]  s3server   0x7200000000000001:0x27  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@4001
    [started]  s3server   0x7200000000000001:0x2a  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1630@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1650
    [started]  hax        0x7200000000000001:0x5b  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@2001
    [started]  ioservice  0x7200000000000001:0x5e  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@3001
    [started]  ioservice  0x7200000000000001:0x6b  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@3002
    [started]  confd      0x7200000000000001:0x78  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@3003
    [started]  s3server   0x7200000000000001:0x7b  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@4001
    [started]  s3server   0x7200000000000001:0x7e  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1650@4002
    cortx-data-headless-svc-ssc-vm-g3-rhev4-2547
    [started]  hax        0x7200000000000001:0x85  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@2001
    [started]  ioservice  0x7200000000000001:0x88  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@3001
    [started]  ioservice  0x7200000000000001:0x95  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@3002
    [started]  confd      0x7200000000000001:0xa2  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@3003
    [started]  s3server   0x7200000000000001:0xa5  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@4001
    [started]  s3server   0x7200000000000001:0xa8  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2547@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1475
    [started]  hax        0x7200000000000001:0xaf  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@2001
    [started]  ioservice  0x7200000000000001:0xb2  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@3001
    [started]  ioservice  0x7200000000000001:0xbf  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@3002
    [started]  confd      0x7200000000000001:0xcc  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@3003
    [started]  s3server   0x7200000000000001:0xcf  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@4001
    [started]  s3server   0x7200000000000001:0xd2  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1475@4002
    cortx-data-headless-svc-ssc-vm-g2-rhev4-1474
    [started]  hax        0x7200000000000001:0xd9  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@2001
    [started]  ioservice  0x7200000000000001:0xdc  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@3001
    [started]  ioservice  0x7200000000000001:0xe9  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@3002
    [started]  confd      0x7200000000000001:0xf6  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@3003
    [started]  s3server   0x7200000000000001:0xf9  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@4001
    [started]  s3server   0x7200000000000001:0xfc  inet:tcp:cortx-data-headless-svc-ssc-vm-g2-rhev4-1474@4002
    cortx-data-headless-svc-ssc-vm-g3-rhev4-2546
    [started]  hax        0x7200000000000001:0x103  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@2001
    [started]  ioservice  0x7200000000000001:0x106  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@3001
    [started]  ioservice  0x7200000000000001:0x113  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@3002
    [started]  confd      0x7200000000000001:0x120  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@3003
    [started]  s3server   0x7200000000000001:0x123  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@4001
    [started]  s3server   0x7200000000000001:0x126  inet:tcp:cortx-data-headless-svc-ssc-vm-g3-rhev4-2546@4002
[root@cortx-data-headless-svc-ssc-vm-g2-rhev4-1473 /]
```

**Pull request merged**